app/views/competitions/_nav.html.erb

```diff
@@ -129,21 +129,32 @@
           }
         end
         unless @competition.results_posted?
-          if @competition.use_wca_registration? && !@competition.uses_new_registration_service?
+          if @competition.use_wca_registration?
             nav_items << {
               text: t('.menu.register'),
               path: competition_register_path(@competition),
               icon: "sign in alt"
             }
           end
-          if @competition.registrations.any? && !@competition.uses_new_registration_service?
+          if @competition.registrations.any?
             nav_items << {
               text: t('.menu.competitors'),
               path: competition_registrations_path(@competition),
-              icon: "users",
+              icon: "users", # Don't add event icons if v2 is used
               tiny_children: @competition.events.map do |event|
-                { text: event.id, path: competition_psych_sheet_event_path(@competition, event.id), cubing_icon: event.id, title: event.name }
-              end
+                if @competition.uses_new_registration_service?
+                  nil
+                else
+                  { text: event.id, path: competition_psych_sheet_event_path(@competition, event.id), cubing_icon: event.id, title: event.name }
+                end
```

Comment:
What about `events.map do |event| ... end unless competition_uses_v2?`?

Reply:
Would save you from explicitly returning nil, and also gets rid of the `.compact` call

Reply:
sadly you can't do `do .. end unless` in an object definition in Ruby it was the first thing I tried as well

Reply:
Then maybe
```
events.map do |event|
  foo if comp.uses_v2_reg?
end.compact
```
could work at least?

Reply:
Making it a variable outside of it though works